packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart

```diff
@@ -222,6 +224,16 @@ class S3Exception extends StorageException {
         recoverySuggestion: AmplifyExceptionMessages.missingExceptionMessage,
       );
 
+  /// Creates a [NetworkException] over a [AWSHttpException] thrown from
+  /// [s3.S3Client] APIs.
+  static NetworkException fromAWSHttpException(AWSHttpException exception) {
+    return NetworkException(
+      'The request failed due to a network error.',
```

Comment:
Should we define a common NetworkException preset as well?  Since the text here is the same as Analytics?

Reply:
Same as my comment above, will make the NetworkException conversion universal in a [separate PR](https://github.com/aws-amplify/amplify-flutter/pull/2781/commits/8925c5f93455136eaadb0c60077e2e46b9933e23).